### PR TITLE
Revert "Use custom CPU thread pool in async_scheduling (#12295)"

### PR DESCRIPTION
**Pull request by gchanan**
This reverts commit 14b48a24045b260d042df24a6f3606526a26ae26.

### CODE REVIEW

**Comment by gchanan** [comment]
Looks like the reverted commit broke the windows build.

**Comment by ilia-cher** [comment]
Windows build was broken because of unrelated to commit issue - linker falls back (because it runs out of memory) to another linker that didn't support a command line option.

**Comment by ilia-cher** [comment]
I'm pretty sure it is a build script issue and not the commit, but I can accept and send a new pr and lets recheck everything

**Comment by gchanan** [comment]
leaving this open, but don't commit because phabricator diff isn't valid.

**Review by ssnl — APPROVED** [review]
If the build script can't be fixed soon, let's unbreak build first.

**Comment by peterjc123** [comment]
@gchanan According to this [post](https://developercommunity.visualstudio.com/content/problem/160714/memory-error-for-linker-in-vs-155-x64.html), this problem can be solved by setting the environmental variable `PreferredToolArchitecture` to `x64`.

**Comment by ezyang** [comment]
I'm a little confused by the discussion here, because the CI results seem to claim the Windows builds passed?

**Comment by peterjc123** [comment]
@ezyang I guess that this PR reverts the change that may cause the aforementioned error. Let's switch the default to the x64 toolchain and then it'll be fine.

**Comment by Orvid** [comment]
```
06:02:02 LINK : the 32-bit linker (C:\PROGRA~2\MIB055~1\2017\COMMUN~1\VC\Tools\MSVC\1411~1.255\bin\Hostx86\x64\link.exe) ran out of heap space; restarting link with the 64-bit linker (C:\Program Files\Git\usr\bin\link.exe)
```
Isn't a build error. The build will succeed, but it will take longer to link, as it's basically trying to link with the 32-bit hosted linker, which fails because the 32-bit address space is too small to be able to link successfully, so it runs the 64-bit hosted linker which then links successfully.

TLDR; The x86_amd64 MSVC toolchain is 32-bit binaries cross-compiling for amd64, and if the linker runs out of memory, it starts the linker from the amd64 (aka. amd64_amd64) toolchain, which is a 64-bit binary and has enough memory to complete.

**Comment by peterjc123** [comment]
@Orvid Yes, you are right. But the alternative x64 linker `C:\Program Files\Git\usr\bin\link.exe` is not what we want here. So #12446 is created.
